ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/DeMarkReversalIndicator.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Marc de Verdelhan & respective authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.pivotpoints;
+
+import org.ta4j.core.Decimal;
+import org.ta4j.core.Tick;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+
+import java.util.List;
+
+/**
+ * DeMark Reversal Indicator.
+ * <p>
+ * @author team172011(Simon-Justus Wimmer), 11.10.2017
+ * @see <a href="http://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:pivot_points">chart_school: pivotpoints</a>
+ */
+public class DeMarkReversalIndicator extends RecursiveCachedIndicator<Decimal> {
+
+    private final DeMarkPivotPointIndicator pivotPointIndicator;
+    private final DeMarkPivotLevel level;
+
+    public enum DeMarkPivotLevel{
```

Comment:
Rename this to "PivotType" and put this enum-declaration within "PivotPointIndicator". With this you can reuse "PivotType" also for other "XPivotPointIndicator" (for example, FibonacciPivotIndicator).

Reply:
DeMark Pivot reversals just have one support and one resistance reversal line